sycl/include/CL/sycl/image.hpp

```diff
@@ -105,38 +194,68 @@ class image {
 
   bool operator!=(const image &rhs) const { return !(*this == rhs); }
 
-  /* -- common interface members -- */
-
   /* -- property interface members -- */
+  template <typename propertyT> bool has_property() const {
+    return impl->template has_property<propertyT>();
+  }
+
+  template <typename propertyT> propertyT get_property() const {
+    return impl->template get_property<propertyT>();
+  }
 
   range<dimensions> get_range() const { return impl->get_range(); }
 
   /* Available only when: dimensions > 1 */
-  range<dimensions - 1> get_pitch() const { return impl->get_pitch(); }
+  template <bool B = (dimensions > 1)>
+  typename std::enable_if<B, range<dimensions - 1>>::type get_pitch() const {
+    return impl->get_pitch();
+  }
 
+  // Returns the size of the image storage in bytes
   size_t get_size() const { return impl->get_size(); }
 
+  // Returns the total number of elements in the image
   size_t get_count() const { return impl->get_count(); }
 
+  // Returns the allocator provided to the image
   AllocatorT get_allocator() const { return impl->get_allocator(); }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::image>
-  get_access(handler &commandGroupHandler) {
-    return impl->template get_access<dataT, accessMode>();
+  //// Provide an accessor to the image with target type image.
+  // template <typename dataT, access::mode accessMode, typename = typename
+  // std::enable_if<detail::is_validImageDataT<dataT>::value,dataT>::type>
+  // accessor<dataT, dimensions, accessMode, access::target::image>
+  // get_access(handler &CommandGroupHandler) {
+  // assert(!"Not Implemented");
+  // return;
+  // // TODO: Define the accessor constructors for images and implement the
+  // APIs.
+  // // return accessor<dataT, dimensions, accessMode, access::target::image,
+  // // access::placeholder::false_t>(*this, CommandGroupHandler);
+  //}
+
+  // template <typename dataT, access::mode accessMode>
+  // accessor<dataT, dimensions, accessMode, access::target::host_image>
+  // get_access() {
+  // assert(!"Not Implemented");
+  // return;
+  // // TODO: Define the accessor constructors for images and implement the
+  // APIs.
+  // // return accessor<dataT, dimensions, accessMode,
+  // access::target::host_image,
+  // // access::placeholder::false_t>(*this);
+  //}
+
+  template <typename Destination = std::nullptr_t>
+  void set_final_data(Destination FinalData = nullptr) {
+    assert(!"Not Implemented");
```

Comment:
I think it's better to throw an exception here.

Reply:
Changed.